gitlab/client.py

```diff
@@ -97,8 +109,11 @@ def __init__(
         self.job_token = job_token
         self._set_auth_info()
 
-        #: Create a session object for requests
-        self.session = session or requests.Session()
+        if http == "requests":
+            self._requests_client = Requests_Client(session=session)
+            self.session = self._requests_client.get_session
+        elif http == "httpx":
+            self._httpx_client = Httpx_Client(client=client)
```

Comment:
Maybe just `self._client` is enough :) as they are mostly API-compatible for making requests.

But I think we will need some more overriding based on multipart-encoding in some of our helper methods. I would try to make a matrix in functional test where we run them with httpx and then you should start seeing some issues with that.

Reply:
I am not sure I know what `multipart-encoding` is

Reply:
Let's see when we start running functional tests with httpx. I think we will run into issues then and we can debug.